LandSandBoat/server
# [DNC] Move logic from 3 abilities to job utils and update their functions to match new effect method

**_I affirm:_**

## What does this pull request do?

Imports some of the changes from PR #3589 

- Reorganizes Dancer job_utils script, so local functions, check functions and use functions are together.
- Moves logic from Building Flourish, Wild Flourish and Contradance to Dancer job_utils
- Updates logic for Flourishes check functions, since we no longer use diferent status effects to count finishing moves, and makes them all use the same one.
- Properly implements Building and Wild Flourishes.

## Steps to test these changes

Better reviewed by commit:

1st - No existing logic changes. Moves code arround inside the same script. Only notable change is renaming a function so it's in line with how the rest are named.
2nd - No existing logic changes. Moves code from ability scripts to job_utils script.
3rd - Updates existing logic in ability check functions, so we only check for 1 status effect and it's power.
4th - Makes all flourishes use the same check function.
5th - Corrects Building and Wild Flourish logic in both job_utils and weaponskills.lua. Corrected the value of Building Flourish merit.
